Common/src/gRPC/Services/Submitter.cs

```diff
@@ -541,20 +541,33 @@ await foreach (var taskRequest in taskRequests.WithCancellation(cancellationToke
     {
       var taskId = Guid.NewGuid()
                        .ToString();
+      var payloadId = Guid.NewGuid()
+                          .ToString();
 
       requests.Add(new TaskCreationRequest(taskId,
-                                           taskId,
+                                           payloadId,
                                            options,
                                            taskRequest.ExpectedOutputKeys.ToList(),
                                            taskRequest.DataDependencies.ToList()));
-      payloadUploadTasks.Add(objectStorage_.AddOrUpdateAsync(taskId,
+      payloadUploadTasks.Add(objectStorage_.AddOrUpdateAsync(payloadId,
                                                              taskRequest.PayloadChunks,
                                                              cancellationToken));
     }
 
     await payloadUploadTasks.WhenAll()
                             .ConfigureAwait(false);
 
+    await resultTable_.Create(requests.Select(request => new Result(sessionId,
```

Comment:
What does this means? Do you change the requirement on whether the payload should be created before hand?

Reply:
The payload is still required to be created in the tasks service.
In the submitter service, payload metadata were not created at all.

Reply:
That was nasty. Fair enough.